lib/nerves_runtime.ex

```diff
@@ -57,16 +53,18 @@ defmodule Nerves.Runtime do
   def revert(opts \\ []) do
     reboot? = if opts[:reboot] != nil, do: opts[:reboot], else: true
 
-    if File.exists?(@revert_fw_path) do
-      {_, 0} = cmd("fwup", [@revert_fw_path, "-t", "revert", "-d", "/dev/rootdisk0"], :info)
+    revert_fw_path = Application.get_env(:nerves_runtime, :revert_fw_path)
```

Comment:
This value can be `nil` so it is safe to provides a fallback value or to ignore when `nil`.

Reply:
It can only be nil if the user configures it as nil. It seems like if someone does that, they'll be able to read the exception and fix their config. The default is unchanged from what it was hardcoded to before.

Reply:
I see. Then maybe `Application.fetch_env!/3` can improve the error message (if I understand correctly).

with `Application.get_env/3`

This might be good enough but the key name is not mentioned.

```elixir
iex> revert_fw_path = nil
nil
iex> File.exists?(revert_fw_path)
** (FunctionClauseError) no function clause matching in IO.chardata_to_string/1

    The following arguments were given to IO.chardata_to_string/1:

        # 1
        nil

    Attempted function clauses (showing 2 out of 2):

        def chardata_to_string(string) when is_binary(string)
        def chardata_to_string(list) when is_list(list)

    (elixir 1.13.4) lib/io.ex:627: IO.chardata_to_string/1
    (elixir 1.13.4) lib/file.ex:216: File.exists?/2
```

with `Application.fetch_env!/3`

This mentions the missing key.

```elixir
iex> Application.fetch_env!(:nerves_runtime, :revert_fw_path)
** (ArgumentError) could not fetch application environment :revert_fw_path for application :nerves_runtime because configuration at :revert_fw_path was not set
    (elixir 1.13.4) lib/application.ex:683: Application.fetch_env!/2
```

Reply:
How are you able to get the key to not exist?

Reply:
I mean the value of `:revert_fw_path ` being `nil`. It might be rare.

Reply:
The reason I'm asking is that I believe that the user has to do this:

```
Application.put_env(:nerves_runtime, :revert_fw_path, nil)
```

or they need to use `:application.unset_env/3` and ignore the warning about not using it. 

`Application.fetch_env!/2` doesn't protect against the first call. The second one seems rare especially for Elixir programmers.

The reason that I'm asking so much is that if you found a way other than those two ways to get a `nil`, I'd like to understand it.

Reply:
One other possibility is a user trying to fetch an environment variable wrongly.

```elixir
# where "MY_FW_PATH" is not set accidentally
config :nerves_runtime, revert_fw_path: System.get_env("MY_FW_PATH")
```

When the use is not aware of the mistake, an error message can be scary and not helpful.

Reply:
Got it. I think this would be a very strange configuration item for someone to call `System.get_env` on, but I understand the sentiment. That's also not caught by `Application.fetch_env!/1`, though. We'd need to add validation code to check that the configuration types are right - i.e., strings and not atoms like `nil` or numbers, etc. 

I know that we have some references to `System.get_env` in Nerves example configurations, but I think using `System.get_env` inside the `config.exs` is an anti-pattern except for passing secrets in CI. And for Nerves devices, passing secrets in this way doesn't hide them since they're stored in the clear in the firmware. I feel like we probably should remove the `System.get_env` calls to avoid the bad examples for new users, but before doing that, lets chat about it since it will affect usability of some examples.

Reply:
I misunderstood `Application.fetch_env!/3`'s behavior. It does not catch `nil` when the key is defined. So in this context, `Application.get_env/3` seems appropriate.